### ISSUE TITLE
Notify chained name for variables

### ISSUE THREAD

**Issue by codingpotato** [SeedV/SeedLang#151]
@wixette Just want to confirm that we really need chained name for variables, for example, `Global.var1`, `Global.func1.var1`. They shall be static scoping name chains. We need implement some mechanism to collect the chained name when variables are defined.

How about class and instance variable after we implement Class? Shall we use `Global.class1.var1` and `Global.func1.object1.var1` as the chained name? Is there any confusion here?

**Comment by codingpotato** [comment]
The static scope doesn't match the dynamic scope. For example, in the recursive call case:
```python
def f(n):
  if n == 1:
    return 1
  else:
    return n * f(n - 1)


f(10)
```
During execution, there will be 10 `n` in the stack. The static scope of them are all f.n. The client still cannot tell which `n` it is? At any time only one local `n` can be access by the client.
@wixette I'm just thinking about if the static scope name is really needed by the client?

**Comment by wixette** [comment]
My original request contains the case of embedded static scopes. For example, in Python:

```
m = 1
class F:
   m = 2
   def foo(self):
     m = 3
     print(m)
```

There are 3 `m` variables in the above code, located at different embedded scopes. We need some mechanism to figure out which `m` is being accessed in visualizers.

If counting Python's embedded functions in, there will be more levels of embedded scopes.

And, also consider my above example:

```
def foo(x, y):
  a = x + y
  return a

def bar(x, y):
  a = x * y
  return a
```

How can we mark variable `a` in different function scopes?

Dynamic scope is not needed to be marked here I thinks. Only static scope is good enough.

**Comment by codingpotato** [comment]
Closed by #186
